run e2e (test level = 2), still fail in 3 cases:
=========================== short test summary info ============================
FAILED testcases/test_collection.py::TestDescribeCollection::test_collection_describe
FAILED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_vectors[get_invalid_vectors3]
FAILED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_vectors[get_invalid_vectors4]
= 3 failed, 2311 passed, 613 skipped, 5 xfailed, 6 xpassed in 4247.61s (1:10:47) =

--- a/internal/core/src/query/GroupByOperator.h
+++ b/internal/core/src/query/GroupByOperator.h
@@ -50,7 +50,11 @@ class GrowingDataGetter : public DataGetter<T> {
 
     T
     Get(int64_t idx) const {
-        return growing_raw_data_->operator[](idx);
+        if constexpr (std::is_same_v<std::string, T>) {
+            return T(growing_raw_data_->view_element(idx));
+        } else {
+            return growing_raw_data_->operator[](idx);
+        }
     }
 };
 

